cordova-lib/src/cordova/util.js

```diff
@@ -401,3 +407,48 @@ function getAvailableNpmVersions(module_name) {
         });
     });
 }
+
+/**
+ * Extracts the name and spec from an npm style package spec that follows the
+ * format of (@scope/)?package(@spec)?
+ * @param {String} raw  The raw npm style package spec
+ * @returns {String[]}  An array containing the package name with optional scope
```

Comment:
IMO returning an array is not very convenient to use. This leads to guessing, what is the 0th item of the result in the expressions like `cordova_util.parseRegistryPluginSpec(target)[0]`. I'd prefer to use structured result. something like `npm-package-arg` returns (https://github.com/npm/npm-package-arg#examples)